### PR TITLE
Makefile: make new step "make stretch" optional

### PR DESCRIPTION
**Pull request by yxieca**
**- What I did**
- Move the old Makefile to secondary make file
- Create a new Makefile to handle extra step

**- How to verify it**
Single thread make is fine. Multiple thread make hits some issue that retry will overcome. Will look into these failures next.

### CODE REVIEW

**Comment by lguohan** [comment]
what is the overhead of make stretch every time?

File: Makefile [hunk 1]
```diff
@@ -1,147 +1,14 @@
...
-	    { echo Image $(SLAVE_IMAGE):$(SLAVE_TAG) not found. Building... ; \
-	    $(DOCKER_BUILD) ; }
-	@$(INSMOD_OVERLAY)
-	@$(DOCKER_RUN) -t $(SLAVE_IMAGE):$(SLAVE_TAG) bash
-
-showtag:
-	@echo $(SLAVE_IMAGE):$(SLAVE_TAG)
-	@echo $(SLAVE_BASE_IMAGE):$(SLAVE_BASE_TAG)
+	@echo "+++ --- Making $@ --- +++"
+	BLDENV=stretch make -f Makefile.work stretch
```
Comment (lguohan) [inline Makefile]: can you add {if endif}, so that we can override the behavior, for example, NOSTRETCH=1 make will not make stretch target implicitly.

**Comment by yxieca** [comment]
Initialing making stretch takes about 90 minutes, most time spent on making kernel. Remake takes about 1 second to confirm nothing needs to be done.

File: Makefile (hunk 1 above)
Comment (yxieca) [reply Makefile]: Sure. Done. Thanks.

File: Makefile [hunk 2]
```diff
@@ -1,147 +1,20 @@
...
-	 if [ $$ans == y ]; then \
-	     git clean -xfdf; \
-	     git reset --hard; \
-	     git submodule foreach --recursive git clean -xfdf; \
-	     git submodule foreach --recursive git reset --hard; \
-	     git submodule update --init --recursive;\
-	 else \
-	     echo "Reset aborted"; \
-	 fi
+clean reset init configure :
```
Comment (lguohan) [inline Makefile]: >clean reset init configure [](start = 0, length = 26)

showtag should be here too.

**Pull request merged**